plugin-api/src/main/java/org/hyperledger/besu/plugin/services/txselection/TransactionSelector.java

```diff
@@ -43,4 +43,17 @@
    */
   TransactionSelectionResult evaluateTransactionPostProcessing(
       PendingTransaction pendingTransaction, TransactionProcessingResult processingResult);
+
+  /**
+   * Method called when a transaction is selected to be added to a block.
+   *
+   * @param pendingTransaction The transaction that has been selected.
+   */
+  default void onTransactionSelected(final PendingTransaction pendingTransaction) {}
+  /**
+   * Method called when a transaction is rejected to be added to a block.
+   *
+   * @param pendingTransaction The transaction that has been rejected.
+   */
+  default void onTransactionRejected(final PendingTransaction pendingTransaction) {}
```

Comment:
Since there are many reasons a tx is not selected for inclusion, it make sense here to also pass the `TransactionSelectionResult` otherwise the plugin could make the wrong assumption.

Then not sure about the word _rejected_, since a tx could not be selected, because it is invalid, so it is also removed from the pool, or just skipped because there is no more space or similarly does not fit with other limits in the current block, but could go in a future block